pkg/build/controller/image_change_controller.go

```diff
@@ -81,16 +80,25 @@ func (c *ImageChangeController) HandleImageRepo(repo *imageapi.ImageRepository)
 
 		if shouldBuild {
 			glog.V(4).Infof("Running build for buildConfig %s in namespace %s", config.Name, config.Namespace)
-			b := buildutil.GenerateBuildFromConfig(config, nil, subs)
-			if err := c.BuildCreator.Create(config.Namespace, b); err != nil {
-				return fmt.Errorf("error starting build for buildConfig %s: %v", config.Name, err)
-			}
+			// first update the config
 			if err := c.BuildConfigUpdater.Update(config); err != nil {
-				// This is not a retryable error because the build has been created.  The worst case
-				// outcome of not updating the buildconfig is that we might rerun a build for the
-				// same "new" imageid change in the future, which is better than running the build
-				// 2+ times by retrying it here.
-				return ImageChangeControllerFatalError{Reason: fmt.Sprintf("Error updating buildConfig %s with new LastTriggeredImageID", config.Name), Err: err}
+				// This is not a retryable error. The worst case outcome of not updating the buildconfig
+				// is that we might rerun a build for the same "new" imageid change in the future,
+				// which is better than guaranteeing we run the build 2+ times by retrying it here.
+				glog.V(2).Infof("Error updating buildConfig %v: %v", config.Name, err)
```

Comment:
the error is getting sent to handleError ultimately, so this is just duplicate logging.